pkg/commands/rules.go

```diff
@@ -305,16 +313,44 @@ func (r *RuleCommand) listRules(k *kingpin.ParseContext) error {
 
 	}
 
-	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.Debug)
+	type namespaceAndRuleGroup struct {
+		Namespace string `json:"namespace" yaml:"namespace"`
+		RuleGroup string `json:"rulegroup" yaml:"rulegroup"`
+	}
+	var items []namespaceAndRuleGroup
 
-	fmt.Fprintln(w, "Namespace\t Rule Group")
 	for ns, rulegroups := range rules {
 		for _, rg := range rulegroups {
-			fmt.Fprintf(w, "%s\t %s\n", ns, rg.Name)
+			items = append(items, namespaceAndRuleGroup{
+				Namespace: ns,
+				RuleGroup: rg.Name,
+			})
 		}
 	}
 
-	w.Flush()
+	switch r.Format {
+	case "json":
+		output, err := json.Marshal(items)
+		if err != nil {
+			return err
+		}
+		fmt.Print(string(output))
+	case "yaml":
+		output, err := yamlv3.Marshal(items)
+		if err != nil {
+			return err
+		}
+		fmt.Print(string(output))
+	default:
+		w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.Debug)
+
+		fmt.Fprintln(w, "Namespace\t Rule Group")
+		for _, item := range items {
+			fmt.Fprintf(w, "%s\t %s\n", item.Namespace, item.RuleGroup)
+		}
+
+		w.Flush()
+	}
```

Comment:
**suggestion[optional]:** It might make sense to move this code to the `pkg/printer` package, and add some unit tests to verify it's output. Let me know if you are interested in doing that and have any questions.

Reply:
Migrated this code to pkg/printer.